src/Formatter/OGMFormatter.php

```diff
@@ -26,12 +30,84 @@ final class OGMFormatter implements FormatterInterface
 {
     public function formatBoltResult(array $meta, iterable $results, Bolt $bolt): Vector
     {
-        throw new BadMethodCallException('Not implemented yet');
+        $results = array_slice($results, 0, count($results) - 1);
+
+        $tbr = new Vector();
+
+        foreach ($results as $result) {
+            $tbr->push($this->formatRow($meta, $result));
+        }
+
+        return $tbr;
     }
 
     public function formatHttpResult(ResponseInterface $response, array $body): Vector
     {
-        throw new BadMethodCallException('Not implemented yet');
+        $tbr = new Vector();
+
+        foreach ($body['results'] as $results) {
+            $tbr->push($this->buildResult($results));
+        }
+
+        return $tbr;
+    }
+
+    private function buildResult(array $result)
+    {
+        $tbr = new Vector();
+
+        $columns = $result['columns'];
+        foreach ($result['data'] as $data) {
+            $row = $data['row'];
+            $meta = $data['meta'];
+            $graph = $data['graph'];
+
+            $record = new Map();
+            foreach ($row as $i => $value) {
+                if (is_array($value)) {
+                    // dealing with properties
+                    switch ($meta[$i]['type']) {
+                        case 'node':
+                            $record->put($columns[$i], Node::makeFromHttpNode($graph['nodes'][$i]));
+                            break;
+                        default:
+                            // @TODO What to do about unsupported types
+                            $record->put($columns[$i], $value);
+                    }
+                } elseif (is_numeric($value)) {
+                    // dealing with a number
+                    $record->put($columns[$i], $value);
+                }
+            }
+
+            $tbr->push($record);
+        }
+
+        return $tbr;
+    }
+
+    private function formatRow(array $meta, array $result): Map
+    {
+        $map = new Map();
+        foreach ($meta['fields'] as $i => $column) {
+            $map->put($column, $this->mapValue($result[$i]));
+        }
+
+        return $map;
+    }
+
+    private function mapValue($value)
+    {
+        if (is_object($value)) {
+            switch (get_class($value)) {
+                case BoltNode::class:
+                    return Node::makeFromBoltNode($value);
+                    break;
+                default:
+                    // @TODO what to do about unsupported types?
```

Comment:
Right now I am thinking about throwing an UnexpectedTypeException with an optional point in the roadmap to make it configurable into a "silent mode", if people need it.

Reply:
I think it would be wiser to make the calling class responsible to check the JSON format and do error handling. This way you can be certain the information coming in is correct. The essence of the problem is making sure you are driving a correct neo4j server. If you are, and the driver is complete, there should not be an error at all.

Reply:
I've never used psalm before. Will look into eliminating the `default` case as soon as i consolidate the factorisation of objects. Maybe that way we won't need a switch/case in the first place.